### PR TITLE
Add takeUpto and dropUpto functions to Data.Vect

### PR DESCRIPTION
**Pull request by locallycompact**
Hi, I thought these would be useful to place here. What do you think?

### CODE REVIEW

File: libs/base/Data/Vect.idr [hunk 1]
```diff
@@ -110,6 +110,21 @@ drop : (n : Nat) -> Vect (n + m) elem -> Vect m elem
 drop Z     xs        = xs
 drop (S k) (x :: xs) = drop k xs
 
+||| Take up to the first n elements of a Vect if they exist.
+||| @ n the maximum number of elements to take
+takeUpto : (n : Nat) -> Vect n elem -> Vect (minimum x n) elem
```
Comment (clayrat) [inline libs/base/Data/Vect.idr]: Duplicate signature

Comment (locallycompact) [reply libs/base/Data/Vect.idr]: Sorry, corrected.

**Comment by locallycompact** [comment]
Fixed another silly error. Any further comments?

**Comment by melted** [comment]
Thanks for the PR!

**Pull request merged**